src/cephadm/cephadm

```diff
@@ -3246,6 +3246,20 @@ def command_inspect_image(ctx):
     return 0
 
 
+def normalize_image_digest(digest):
+    # normal case:
+    #   ceph/ceph -> docker.io/ceph/ceph
+    # edge cases that shouldn't ever come up:
+    #   ubuntu -> docker.io/ubuntu    (ubuntu alias for library/ubuntu)
+    # no change:
+    #   quay.ceph.io/ceph/ceph -> ceph
+    #   docker.io/ubuntu -> no change
+    bits = digest.split('/')
+    if '.' not in bits[0] or len(bits) < 3:
+        digest = 'docker.io/' + digest
```

Comment:
this is actually not totally true. this depends on the [`unqualified-search-registries`](https://github.com/sebastian-philipp/registries-conf-ctl/blob/40c90b590324573b69052862d372f683e184bd53/registries_conf_ctl/cli.py#L72) parameter of the registries.conf and might vary between hosts.

Reply:
Hmm.  When would it not be docker.io though?  On my dev host, for instance, ``selenium/standalone-chrome:3.141`` has repo digest ``selenium/standalone-chrome@sha256:8d0915d29d5fa495256bda78371cb5adb70fd16ec9e4b4aaae72a0181682f5f0``.  But registries.conf has
```
unqualified-search-registries = ["registry.fedoraproject.org", "registry.access.redhat.com", "registry.centos.org", "docker.io"]
```

It seems like we could only have an ambiguous situation come up if we have an unqualified image name in one of the search registries. Perhaps we should, in addition to this patch, also require fully-qualified image names for the ``ceph orch upgrade start'`` command, and/or canonicalize the image name that we get from ``container_image`` so that we'll always have the docker.io prefix (or some other registry).

I'm not sure if it's possible, though, that we pull something from registry foo (not docker.io) and inspect it and see a RepoDigest that is unqualified (and therefore incorrectly prepend docker.io). We could instead canonicalize using the registry we just pulled from... maybe that is the safest thing?